[FIX] CI 태스크 변경

jye326 opened the pull request:
## Issue Number
closed #640 

## As-Is
<!-- 문제 상황 정의 -->

- `build-and-push` 에서 -x test를 제거하여 test를 실행합니다.
- 기존 : PR 생성 시 `backend-ci` : 테스트 실행 -> PR 머지 시 `build-and-push` : 테스트 X, 도커이미지 빌드 후 도커 허브에 배포 -> `deploy-dev` : 도커 허브에서 pull , 배포 스크립트 실행

## To-Be
<!-- 변경 사항 -->

- rel-dev 브랜치로의 PR 생성을 트리거로 동작하는 `backend-ci` 태스크를 제거합니다.
- 변경 : PR 생성 시 작업 X -> PR 머지 시 `build-and-push` : 테스트 실행, 도커이미지 빌드 후 도커 허브에 배포 -> `deploy-dev` : 도커 허브에서 pull, 배포 스크립트 실행

- 이제 도커 이미지 빌드 전에 test가 돌아가 api 문서 자동 배포 태스크가 성공합니다.

## 이슈

> - 빌드하는 도커 이미지 파일이 스니펫을 위한 테스트 실행을 전제하기 때문에 `build-and-push`에서 테스트를 실행해야 합니다
> `backend-ci`는 발동 조건이 pr이라 pr에 변경사항이 일어날 때마다 테스트를 수행합니다.
> 배포 워크 플로우 이전에 develop 브랜치에서 테스트가 이미 수행된 상태로 배포 PR이 생성되기 때문에 `backend-ci` 태스크가 불필요하다고 판단하여 제거하였습니다.


## Check List
- [x] 테스트가 전부 통과되었나요?
- [x] 모든 commit이 push 되었나요?
- [x] merge할 branch를 확인했나요?
- [x] Assignee를 지정했나요?
- [x] Label을 지정했나요?
- [x] 닫을 이슈 번호를 지정했나요?

jye326 committed:
delete: backend-ci 워크플로우 제거

ppparkta reviewed (commented):
고생하셨습니다. CI가 `./gradlew clean generateApiDocs bootJar` 작업 내에 포함되어서 앞 단계가 지워진걸까요?

1. test 실행하기 위해서 test 전용 환경변수가 필요한 것으로 알고있는데 이렇게 태스크가 변경돼도 정상적으로 실행 되나요? 현재 환경변수 지정해주는 부분이 안 보여서요!

2. 더불어 test할 때 사용할 MySQL을 컨테이너로 띄우는 작업도 없어져서 테스트 빌드가 안 될 것으로 보여요! 제가 이해한 것과 다른 방식으로 동작하고 있나요?!

jye326 committed:
add: backend-ci 워크플로우 추가 (release-prod, release-dev)

jye326 committed:
feat: CI/CD workflows에 테스트 DB 설정 및 API 문서 생성 태스크 추가

- Docker Compose로 테스트 MySQL 컨테이너 실행 및 대기 로직 추가
- Gradle 빌드 전 테스트 환경 변수 설정을 위한 .env 파일 생성 로직 추가
- gradlew 실행 권한 부여 및 API 문서 생성 포함한 빌드 태스크 구성
- 테스트 DB 컨테이너 종료 로직 추가

jye326 committed:
delete: backend-ci 워크플로우 제거 (release-prod, release-dev)

kimdevv commented on .github/workflows/release-prod-ci-cd.yml:
```diff
@@ -10,22 +10,29 @@ on:
+      - uses: actions/checkout@v4
 
-      - name: Set up Docker Compose
+      - name: Set up Docker Compose & nc
```
`nc`는 어떤 걸 하는 건가요?

kimdevv commented on .github/workflows/release-prod-ci-cd.yml:
```diff
@@ -34,10 +41,12 @@ jobs:
         run: |
           echo "Waiting for Test MySQL..."
           for i in {30..0}; do
-            nc -z localhost 3307 && echo "Test MySQL is up!" && break
+            nc -z localhost 3307 && echo "Test MySQL is up!" && exit 0
```
`exit 0`과 `break`는 어떤 차이인가요??

kimdevv reviewed (approved):
고생하셨습니다~

ppparkta reviewed (approved):
feature -> develop 병합 시 여태까지 했던대로 CI만 수행되고,
develop -> release-dev/prod 병합 시 CI(rest docs 스니펫 생성 포함) + CD 수행되는 로직인 것 같네요!

이렇게 하면 RestDocs 확인 문제를 해결할 수 있을 것 같아요 고생하셨어요!
